Fix: Removing base64 encoding (#534)
codemagic-ci-cd/codemagic-docs: content/knowledge-base/load-firebase-configuration.md

  - '../custom-scripts/load-firebase-configuration'
---

Instead of committing the Firebase configuration files to your repository, you can upload them to Codemagic as **environment variables** and reference them in a custom script.

> 1.  Save your copy-pasted contents of Firebase config files as environment variables, e.g. `ANDROID_FIREBASE_SECRET` and `IOS_FIREBASE_SECRET` in Codemagic UI (either in Application/Team variables) and select **secure**. 

Reviewer: ```suggestion
1.  Save the **contents** of Firebase config files as environment variables, e.g. `ANDROID_FIREBASE_SECRET` and `IOS_FIREBASE_SECRET` in Codemagic UI (either in Application or Team variables) and select **Secure**. 
```

  - '../custom-scripts/load-firebase-configuration'
---

> Instead of committing the Firebase configuration files to your repository, you can upload them to Codemagic as [environment variables](../building/environment-variables/) and reference them in a custom script. Note that the Firebase configuration file (`google-services.json` for Android or `GoogleService-Info.plist` for iOS) must be selected as **secure**.

Reviewer: It's better to replace the path to environment variables with the new path (although the one here will probably also work). I would completely remove the last sentence here.

> 1.  Save your Firebase config files as environment variables, e.g. `ANDROID_FIREBASE_SECRET` and `IOS_FIREBASE_SECRET`. 

Reviewer: Here I would mention the need to copy-paste the contents of the file to save them to env vars and clicking Secure to encrypt them.

  PROJECT_ROOT=$FCI_BUILD_DIR/myproject/path      # ADD YOUR PROJECT FOLDER PATH HERE

  echo $ANDROID_FIREBASE_SECRET > $PROJECT_ROOT/android/app/google-services.json
  echo $IOS_FIREBASE_SECRET > $PROJECT_ROOT/ios/Runner/GoogleService-Info.plist
  ```

> For more details about loading the Firebase configuration in Codemagic, see the step-by-step guide [here](https://blog.codemagic.io/how-to-load-firebase-config-in-codemagic-with-environment-variables/).

Reviewer: can you also confirm the reference here is up to date? if it's not, let's remove it

Author: All changes done!
